src/charm.py

```diff
@@ -242,6 +242,36 @@ def _get_external_hostname(self) -> str:
             else self.app.name
         )
 
+    def _get_cors_origin(self) -> str:
+        """Return the combined CORS origins.
+        
+        Return the combined CORS origins from 'cors_origin' and, if enabled,
+        'external_hostname' and 's3_cdn_url'. Skips augmentation if 'cors_origin' is '*'.
+
+        Returns:
+            Comma-separated CORS origins string.
+        """
+        user_value = self.config.get("cors_origin", "").strip()
+        if user_value == "*":
+            # No need to augment if all origins allowed
+            return "*"
+
+        origins = set()
+        if user_value:
+            origins.update(o.strip() for o in user_value.split(",") if o.strip())
```

Comment:
Minor suggestion, feel free to reject, or use a changed version:
```suggestion
            origins.update(s for s in (o.strip() for o in user_value.split(","))if s)
```
This calls `o.strip()` once. `s` stands for stripped.